#! /usr/bin/env python

PACKAGE='image_cb_detector'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                  Type      Level   Description                   Default   Min   Max
gen.add("num_x",              int_t,    0,      "Number of columns",          8,        1,    30)
gen.add("num_y",              int_t,    0,      "Number of rows",             6,        1,    30)
gen.add("dim_x",              double_t, 0,      "Space between columns",      1,        0,    1)
gen.add("dim_y",              double_t, 0,      "Space between rows",         1,        0,    1)
gen.add("min_area",           double_t, 0,      "Minimum area of a circle (in px)",   25,       1,    10000)
gen.add("max_area",           double_t, 0,      "Maximum area of a circle (in px)",   5000,     1,    10000)
gen.add("min_dist",           double_t, 0,      "Minimum distance between circles (in px)", 10, 1, 1000)
gen.add("clustering",         bool_t,   0,      "Use a special algorithm for grid detection",  False)

exit(gen.generate(PACKAGE, "image_cb_detector", "Circles"))
